xline/src/client/mod.rs

```diff
@@ -0,0 +1,183 @@
+use std::net::SocketAddr;
+
+use anyhow::{anyhow, Result};
+use curp::{client::Client as CurpClient, cmd::ProposeId};
+use etcd_client::Client as EtcdClient;
+use prost::Message;
+use uuid::Uuid;
+
+use crate::{
+    rpc::{
+        DeleteRangeResponse, PutResponse, RangeResponse, Request, RequestOp, Response, ResponseOp,
+    },
+    server::command::{Command, KeyRange},
+};
+
+use kv_types::{PutRequest, RangeRequest};
+
+use self::kv_types::DeleteRangeRequest;
+
+/// covert struct between etcd and curp
+mod convert;
+/// Requests used by Client
+pub mod kv_types;
+
+/// Xline client
+#[allow(missing_debug_implementations)] // EtcdClient doesn't implement Debug
+pub struct Client {
+    /// Name of the client
+    name: String,
+    /// Curp client
+    curp_client: CurpClient<Command>,
+    /// Etcd client
+    etcd_client: EtcdClient,
+    /// Use curp client to send requests when true
+    use_curp_client: bool,
+}
+
+impl Client {
+    /// New `Client`
+    ///
+    /// # Panics
+    ///
+    /// Panics if `EtcdClient::connect` fails.
+    #[inline]
+    pub async fn new(
+        leader_index: usize,
+        all_members: Vec<SocketAddr>,
+        use_curp_client: bool,
+    ) -> Self {
+        let etcd_client = EtcdClient::connect(
+            all_members
+                .iter()
+                .map(ToString::to_string)
+                .collect::<Vec<_>>(),
+            None,
+        )
+        .await
+        .unwrap_or_else(|e| panic!("failed to connect to server: {}", e));
```

Comment:
Client should not panic if any connection fails, even returning an error is a better solution.

Reply:
We'd better create one kind of client, either etcd or curp, and providing an argument to let user choose a client.

Reply:
we need both of them, because some requests like put can use curp client get a faster response, but requests like watch can not use curp client , they communicate directly with the server use etcd client